packages/components/addon/components/hds/tabs/index.js

```diff
@@ -16,26 +16,37 @@ export default class HdsTabsIndexComponent extends Component {
   @tracked panelIds = [];
   @tracked selectedTabIndex;
 
+  get selectedIndex() {
+    return this.selectionControlled
+      ? this.args.selectedTabIndex
+      : this.selectedTabIndex;
+  }
+
+  get selectionControlled() {
+    return this.args.selectedTabIndex >= 0;
+  }
+
   @action
   didInsert() {
     // default starting tab index
     let initialTabIndex = 0;
     let selectedCount = 0;
 
-    this.tabNodes.forEach((tabElement, index) => {
-      if (tabElement.hasAttribute('data-is-selected')) {
-        initialTabIndex = index;
-        selectedCount++;
-      }
-    });
-    this.selectedTabIndex = initialTabIndex;
+    if (!this.selectionControlled) {
+      this.tabNodes.forEach((tabElement, index) => {
+        if (tabElement.hasAttribute('data-is-selected')) {
+          initialTabIndex = index;
+          selectedCount++;
+        }
+      });
+      this.selectedTabIndex = initialTabIndex;
+    }
 
     schedule('afterRender', () => {
-      this.setTabIndicator(initialTabIndex);
+      this.setTabIndicator();
```

Comment:
`did-update` does not fire when the component is initially added so we need to fire this here to set the indicator on render.